File: leetcode/java/771. Jewels and Stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        final Map<Character, Object> store = new HashMap<>();
        
        for(int i = 0; i < jewels.length(); i++) {
            store.put(jewels.charAt(i), new Object());
        }
        
        int ret = 0;
        for(int i = 0; i < stones.length(); i++) {
            final char stone = stones.charAt(i);
            
            ret += (store.get(stone) == null) ? 0 : 1;
        }
        
        return ret;
    }
}

File: leetcode/kotlin/771. Jewels and Stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        val store = mutableMapOf<Char, Object>();
        
        for(index in 0 until jewels.length) {
            store.put(jewels[index], Object())
        }
        
        var ret = 0
        for(index in 0 until stones.length) {
            ret += if(store.get(stones[index]) == null) 0 else 1
        }
        
        return ret
    }
}
